core/bin/external_node/src/main.rs

```diff
@@ -461,6 +627,56 @@ struct Cli {
     /// This is an experimental and incomplete feature; do not use unless you know what you're doing.
     #[arg(long)]
     enable_snapshots_recovery: bool,
+    /// Comma-separated list of components to launch.
+    #[arg(long, default_value = "all")]
+    components: ComponentsToRun,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Hash, Eq)]
+pub enum Component {
+    HttpApi,
+    WsApi,
+    Tree,
+    TreeApi,
+    Core,
+}
+
+impl Component {
+    fn components_from_str(s: &str) -> anyhow::Result<&[Component]> {
+        match s {
+            "api" => Ok(&[Component::HttpApi, Component::WsApi]),
+            "http_api" => Ok(&[Component::HttpApi]),
+            "ws_api" => Ok(&[Component::WsApi]),
+            "tree" => Ok(&[Component::Tree]),
+            "tree_api" => Ok(&[Component::TreeApi]),
+            "core" => Ok(&[Component::Core]),
+            "all" => Ok(&[
+                Component::HttpApi,
+                Component::WsApi,
+                Component::Tree,
+                Component::Core,
+            ]),
+            other => Err(anyhow::anyhow!("{} is not a valid component name", other)),
```

Comment:
Nit: You can specify `other` in an interpolated string: `{other} is not a valid component name`.